pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1418,12 +1418,13 @@ public synchronized PulsarClient getClient() throws PulsarServerException {
                 ClientConfigurationData conf =
                         ConfigurationDataUtils.loadData(overrides, initialConf, ClientConfigurationData.class);
 
-                conf.setServiceUrl(this.getConfiguration().isTlsEnabled()
-                                ? this.brokerServiceUrlTls : this.brokerServiceUrl);
-                conf.setTlsAllowInsecureConnection(this.getConfiguration().isTlsAllowInsecureConnection());
-                conf.setTlsTrustCertsFilePath(this.getConfiguration().getTlsCertificateFilePath());
+                boolean tlsEnabled = this.getConfiguration().isBrokerClientTlsEnabled();
```

Comment:
Looks like the reason why we can't use the `isTlsEnabled` here is that we can have tls port and non-tls port for a broker? It's better to add a description here to clarify it.

And, please also add the compatibility explanation for the change. Or do we need to consider the compatibility issue? Since users can upgrade to a new version broker but with the old version config file.

The document of `brokerClientTlsEnabled` also needs to update since we also use broker clients for system topics.

Reply:
> Looks like the reason why we can't use the `isTlsEnabled` here is that we can have tls port and non-tls port for a broker? It's better to add a description here to clarify it.

The `isTlsEnabled` has been deprecated and used for the broker, not the client.  The `brokerClient` prefix is used for client/admin in config file.

Currently, the `getAdminClient()` has the same check, link to https://github.com/apache/pulsar/pull/16014/files#diff-8db8be87611ed21e4a80eb15b8749729d23a49a835cb0401c8c72e5225241b21R1461, we should keep config.

> And, please also add the compatibility explanation for the change. Or do we need to consider the compatibility issue? Since users can upgrade to a new version broker but with the old version config file.

OK, we should update this description to be cleaner.

Reply:
@nodece Could you please provide more details of `No breaking changes.`? Without this change, `this.getConfiguration().isTlsEnabled()` will be used to create the broker client. But after this change, users must configure `brokerClientTlsEnabled`, what will happen if `brokerClientTlsEnabled` absents?

Reply:
Currently, there has a bug:
```
conf.setTlsTrustCertsFilePath(this.getConfiguration().getTlsCertificateFilePath());
```
The `getTlsCertificateFilePath()` is not ca cert, which shoule be `this.getConfiguration().getTlsCertificateFilePath()` or `this.getConfiguration().getBrokerClientTrustCertsFilePath()`.

When `this.getConfiguration().isTlsEnabled()` is `true`, users will meet `Failed reason: General OpenSslEngine problem`.

Reply:
Is it worth making it like this:

```suggestion
                boolean tlsEnabled = this.getConfiguration().isBrokerClientTlsEnabled() || this.getConfiguration().isTlsEnabled();
```

That would ensure backwards compatibility.

Reply:
@nodece - is your point that since `tlsEnabled` is broken, we don't need to support it?

Reply:
> @nodece - is your point that since `tlsEnabled` is broken, we don't need to support it?

Yes, I want to using `this.getConfiguration().isBrokerClientTlsEnabled()` for keep same with create admin client.

Reply:
Thanks @michaeljmarshall.